### PR TITLE
add engine_getClientVersionV1

### PR DESCRIPTION
**Pull request by Matilda-Clerke**
## PR description
Add the engine_getClientVersionV1 to the engine RPC API
## Fixed Issue(s)

https://github.com/hyperledger/besu/issues/6471

### CODE REVIEW

File: ethereum/api/src/integration-test/java/org/hyperledger/besu/ethereum/api/jsonrpc/JsonRpcTestMethodsFactory.java [hunk 1]
```diff
@@ -64,7 +64,9 @@
 /** Provides a facade to construct the JSON-RPC component. */
 public class JsonRpcTestMethodsFactory {
 
-  private static final String CLIENT_VERSION = "TestClientVersion/0.1.0";
+  private static final String CLIENT_NODE_NAME = "TestClientVersion/0.1.0";
```
Comment (Matilda-Clerke) [inline ethereum/api/src/integration-test/java/org/hyperledger/besu/ethereum/api/jsonrpc/JsonRpcTestMethodsFactory.java:67]: Renaming this to better match its source in BesuInfo

File: besu/src/main/java/org/hyperledger/besu/BesuInfo.java [hunk 2]
```diff
@@ -27,6 +29,21 @@ public final class BesuInfo {
   private static final String VERSION = BesuInfo.class.getPackage().getImplementationVersion();
   private static final String OS = PlatformDetector.getOS();
   private static final String VM = PlatformDetector.getVM();
+  private static final String COMMIT;
+
+  static {
+    if (VERSION == null) {
+      COMMIT = null;
```
Comment (usmansaleem) [inline besu/src/main/java/org/hyperledger/besu/BesuInfo.java:36]: instead of NULL, would it be better to return empty String `""`?

Comment (usmansaleem) [reply besu/src/main/java/org/hyperledger/besu/BesuInfo.java:36]: Also, is there a test that covers NULL commit? How would it be reported?

Comment (Matilda-Clerke) [reply besu/src/main/java/org/hyperledger/besu/BesuInfo.java:36]: Unfortunately, it's difficult to test the value of COMMIT because BesuInfo pulls data from the class' package which isn't typically set outside of a proper build.

As for null vs empty String, philosophically I'd rather it be null. The api doesn't really specify what should be supplied if a value isn't available.

File: besu/src/main/java/org/hyperledger/besu/BesuInfo.java [hunk 3]
```diff
@@ -27,6 +29,21 @@ public final class BesuInfo {
...
+    } else {
+      Pattern pattern = Pattern.compile("v?\\d*\\.\\d*\\.\\d*-\\w+-(?<commit>[0-9a-fA-F]{8})");
+      Matcher matcher = pattern.matcher(VERSION);
```
Comment (jframe) [inline besu/src/main/java/org/hyperledger/besu/BesuInfo.java:39]: Only the develop builds include the commit hash in the version at the moment, so this won't work for github release versions.

I tested locally using `./gradlew -Prelease.releaseVersion=99.9.0 -Pversion=99.9.0 assemble installDist` and the commit field in the response was null.

Comment (Matilda-Clerke) [reply besu/src/main/java/org/hyperledger/besu/BesuInfo.java:39]: Yeah, I'm not sure how else to get the commit hash. The purpose of the field is to give a specific commit at which the application was run, and for github releases there seems to be an accompanying tag, so the version itself can be used to find the exact commit if desired.

Do you think it would be worthwhile finding some other way to get the commit field populated?

**Review by pinges — APPROVED** [review]
LGTM
